src/ConsentRequiredController.js

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint max-len: [2, 160] */
+define([
+  'okta',
+  'shared/util/Util',
+  'util/FormController',
+  'util/FormType',
+  'views/consent/ScopeList'
+],
+function (Okta, Util, FormController, FormType, ScopeList) {
+
+  var _ = Okta._;
+
+  return FormController.extend({
+    className: 'consent-required',
+    initialize: function () {
+      this.model.set('expiresAt', this.options.appState.get('expiresAt'));
+      this.model.set('scopes', this.options.appState.get('scopes'));
+      this.listenTo(this.form, 'save', _.bind(this.model.save, this.model));
+      this.listenTo(this.form, 'cancel', _.bind(this.model.cancel, this.model));
+    },
+    Model: {
+      props: {
+        expiresAt: ['string', true],
+        scopes: ['array', true]
+      },
+      save: function () {
+        return this.doTransaction(function(transaction) {
+          return transaction.consent({
+            consent: {
+              expiresAt: this.get('expiresAt'),
+              scopes: _.pluck(this.get('scopes'), 'name')
+            }
+          });
+        });
+      },
+      cancel: function () {
+        var self = this;
+        return this.doTransaction(function(transaction) {
+          return transaction.cancel();
+        }).then(function () {
+          self.settings.get('consent').cancel();
+        });
+      }
+    },
+    Form: {
+      noCancelButton:false,
```

Comment:
default is false?

Reply:
default is true in signin widget, in particular in `FormController`